lib/app/client.js

```diff
@@ -292,11 +292,22 @@ async function render (to, from, next) {
 
     // Call .validate()
     let isValid = true
-    Components.forEach((Component) => {
-      if (!isValid) return
-      if (typeof Component.options.validate !== 'function') return
+    let validateReturn
+    for (const Component of Components) {
+      if (typeof Component.options.validate !== 'function') {
+        continue
+      }
+
       isValid = Component.options.validate(app.context)
-    })
+      validateReturn = await Component.options.validate(app.context)
```

Comment:
I think await is fine, return value shouldn’t be nested promise

Reply:
And await works fine with non-async method

Reply:
> I think await is fine, return value shouldn’t be nested promise

The goal here is to support functions, async functions and functions that return promise (which are essentially the same as async functions) the same way. If a promise is returned, we need to await on it, if not, we can use the return value as is.

Reply:
They all work with await I think

Reply:
Return promise is same as async function.
Non async function will aotu append promise.resolve on return value if use await

Reply:
@clarkdo cool, I've updated it.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#Description